### PR TITLE
[communication-email] Addressing SDK arch board feedback

### PR DESCRIPTION
**Pull request by yogeshmo**
# Description

- AAD token authentication was added to the client
- The SDK is now model-less

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-email/README.md [hunk 1]
```diff
@@ -39,6 +39,17 @@ connection_string = "endpoint=https://<resource-name>.communication.azure.com/;a
 client = EmailClient.from_connection_string(connection_string);
 ```
 
+Alternatively, you can also use Active Directory authentication using DefaultAzureCredential.
+
+```python
+from azure.communication.email import Email
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/README.md:45]:
```suggestion
from azure.communication.email import EmailClient
```

File: sdk/communication/azure-communication-email/azure/communication/email/_email_client.py [hunk 2]
```diff
@@ -63,16 +68,86 @@ def from_connection_string(
...
+                          message.
+                    },
+                    "recipients": {
+                        "to": [
+                            {
+                                "email": "str",  # Email address. Required.
+                                "displayName": "str"  # Optional. Email display name.
+                            }
+                        ],
+                        "CC": [
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/_email_client.py:100]: Does the service treat these as case insensitive?

File: sdk/communication/azure-communication-email/README.md [hunk 3]
```diff
@@ -144,7 +162,7 @@ The result from the `send` call contains a `message_id` which can be used to que
 
 ```python
 response = client.send(message)
-status = client.get_sent_status(response.message_id)
+status = client.get_sent_status(response['message_id'])
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/README.md:165]:
```suggestion
status = client.get_sent_status(response['messageId'])
```

File: sdk/communication/azure-communication-email/azure/communication/email/_generated/_client.py [hunk 4]
```diff
@@ -36,40 +34,28 @@ class AzureCommunicationEmailService(object):  # pylint: disable=client-accepts-
...
-
+        self.email = EmailOperations(self._client, self._config, self._serialize, self._deserialize)
 
-    def _send_request(
-        self,
-        request,  # type: HttpRequest
-        **kwargs  # type: Any
-    ):
-        # type: (...) -> HttpResponse
+    def send_request(self, request: HttpRequest, **kwargs: Any) -> HttpResponse:
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/_generated/_client.py:49]: We can expose this in the EmailClient as well.

File: sdk/communication/azure-communication-email/azure/communication/email/_generated/operations/_patch.py [hunk 5]
```diff
@@ -6,59 +6,71 @@
...
+
+if sys.version_info >= (3, 9):
+    from collections.abc import MutableMapping
+else:
+    from typing import MutableMapping  # type: ignore  # pylint: disable=ungrouped-imports
+JSON = MutableMapping[str, Any]  # pylint: disable=unsubscriptable-object
 
 class EmailOperations(EmailOperationsGenerated):
 
     def __return_message_id(self, pipeline_response, _, response_headers):
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/_generated/operations/_patch.py:21]: Just a single underscore is the convention for a private method.

File: sdk/communication/azure-communication-email/azure/communication/email/_generated/operations/_patch.py [hunk 6]
```diff
@@ -6,59 +6,71 @@
...
+            repeatability_request_id=repeatability_request_id,
+            repeatability_first_sent=repeatability_first_sent,
+            content_type=content_type,
+            cls=self.__return_message_id,
+            **kwargs
         )
-        return SendEmailResult(message_id=message_id)
-    
+
+        return { "message_id": message_id }
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/_generated/operations/_patch.py:69]: Looks like we should make this `{"messageId": message_id}` to be consistent with the response from `get_status`

File: sdk/communication/azure-communication-email/azure/communication/email/_generated/aio/operations/_operations.py [hunk 7]
```diff
@@ -31,254 +43,298 @@ class EmailOperations:
...
-        response_headers['Repeatability-Result']=self._deserialize('str', response.headers.get('Repeatability-Result'))
-        response_headers['Operation-Location']=self._deserialize('str', response.headers.get('Operation-Location'))
-        response_headers['Retry-After']=self._deserialize('int', response.headers.get('Retry-After'))
-        response_headers['x-ms-request-id']=self._deserialize('str', response.headers.get('x-ms-request-id'))
-
+        response_headers["Repeatability-Result"] = self._deserialize(
+            "str", response.headers.get("Repeatability-Result")
+        )
+        response_headers["Operation-Location"] = self._deserialize("str", response.headers.get("Operation-Location"))
+        response_headers["Retry-After"] = self._deserialize("int", response.headers.get("Retry-After"))
```
Comment (annatisch) [inline sdk/communication/azure-communication-email/azure/communication/email/_generated/aio/operations/_operations.py:336]: It looks like on the service this was implemented as a long running async operation?

File: sdk/communication/azure-communication-email/azure/communication/email/_generated/_client.py (hunk 4 above)
Comment (annatisch) [reply sdk/communication/azure-communication-email/azure/communication/email/_generated/_client.py:49]: Actually may let's not for now - I don't want it to get confused with the existing `send` method on the client.

**Comment by yogeshmo** [comment]
Still active

File: sdk/communication/azure-communication-email/azure/communication/email/_email_client.py (hunk 2 above)
Comment (yogeshmo) [reply sdk/communication/azure-communication-email/azure/communication/email/_email_client.py:100]: Good catch. The service needs these lowercase, I'll change this.

**Comment by petrsvihlik** [comment]
@yogeshmo could you please consider opening up a new PR? this one is impossible to review
image

also there seem to be changes in `azure-ai-ml` which I'm not sure are intended:
image

**Pull request closed without merging**